### DISCUSSION TITLE
After updating to version 12,  PurchaseState is always 1 (Canceled) on server validation

### DISCUSSION THREAD

**Discussion by trickeyd** [hyochan/react-native-iap#2245] [Q&A]
Hello - great lib - thanks for you r hard work.

I recently upgraded our react native version from 0.66 -> 0.71, and I upgraded this lib at the same time. After the migration, everything seemed to be working fine, and iOS has no issues, but when making transactions on Android, the Play Store returns a 'Canceled' status for all transactions, so they can never complete.

I know it's due to the upgrade because going back to the old version on the same server still works, but I can see nothing in the purchase received by purchaseUpdatedListener that looks any different.

I have been fiddling with it all day so any pointers in the right direction would be greatly appreciated!

**Comment by trickeyd** [comment]
I ended up downgrading to 11.0.7 and both platforms are working again. Any idea why this could be?

**Comment by andresesfm** [comment]
Hard to tell given the provided information. Did you see any errors when finishing the transactions?
